test/fw/ptl/utils/pbs_anonutils.py

```diff
@@ -98,6 +100,96 @@ def __init__(self, attr_delete=None, resc_delete=None,
         self.gmap_attr_key = {}
         self.gmap_resc_key = {}
 
+    def __get_kvl(self, text):
+        """
+        Extract key-value pairs from the text specified
+
+        :param text: the line of text to extract key-value pairs from
+        :type text: String
+
+        :returns: a dict of key value pairs
+        """
+        if text is None:
+            return None
+
+        if "=" not in text:
+            # No key-value pairs in the text
+            return None
+
+        # Split the text by spaces
+        list_items = shlex.split(text)
+
+        kvl = {}
+
+        # Go through the list and handle the following two cases:
+        #     "key = value"
+        #     "key=value"
+        index = 0
+        for item in list_items:
```

Comment:
I think this is one of the cases you want to do a for index in range(len(list_items)):
The way you're doing it, you are making an assumption where the item is in the list based on your index variable.  It'd be safer to just be able to index into list_items.